gwsumm/html/bootstrap.py

```diff
@@ -109,9 +110,18 @@ def calendar(date, tag='a', class_='navbar-brand dropdown-toggle',
     page = markup.page()
     page.a('&laquo;', class_='navbar-brand step-back', title='Step back',
            onclick='stepDate(-1)')
+    attributekwargs = {'data-date': data_date,
+                       'data-date-format': 'dd-mm-yyyy',
+                       'data-viewmode': '%ss' % mode.name}
+    if highlighteddates is not None:
+        attributekwargs['highlight-dates'] = highlighteddates.replace('-', '')
+    elif highlightavailable:
+        # set location of dates file
+        datefile = '{}/list-dirs.txt'.format(
```

Comment:
@mattpitkin, see my comment to `gw_summary` above. If we decide to keep this file rather than the raw list of dates, then this variable should actually be passed as a keyword argument with a default of `None`, i.e.

```python
def calendar(date, tag='a', class_='navbar-brand dropdown-toggle',
             id_='calendar', dateformat=None, mode=None, datefile=None,
             highlighteddates=None, highlightavailable=False):
```

That would make this a lot more robust as it would enforce that the file expected here be the same as the one defined in `gw_summary`. Any calls to `gwsumm.html.bootstrap.calendar` would have to be updated along with this function's docstring and unit tests.

Reply:
I've made changes to pass a `datefile` argument as suggested. In `gw_summary` I get it to add the file name to the `config` variable.